pallets/asset/src/lib.rs

```diff
@@ -2245,6 +2245,12 @@ impl<T: Trait> Module<T> {
         Ok(())
     }
 
+    fn archive_error_selector(is_archive: bool) -> Error<T> {
```

Comment:
Now the closure is a function that is used once, but its still used once. It doesn't add value, let's inline it.